### PR TITLE
8273372: Remove scavenge trace message in psPromotionManager

### PR DESCRIPTION
**Pull request by albertnetymk**
Simple change of moving `ResourceMark` closer to where it is actually required.

Test: tier1-6

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273372](https://bugs.openjdk.java.net/browse/JDK-8273372): Remove scavenge trace message in psPromotionManager

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)
 * [Leo Korinth](https://openjdk.java.net/census#lkorinth) (@lkorinth - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5375/head:pull/5375` \
`$ git checkout pull/5375`

Update a local copy of the PR: \
`$ git checkout pull/5375` \
`$ git pull https://git.openjdk.java.net/jdk pull/5375/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5375`

View PR using the GUI difftool: \
`$ git pr show -t 5375`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5375.diff">https://git.openjdk.java.net/jdk/pull/5375.diff</a>

</details>

### CODE REVIEW

**Review by tschatzl — APPROVED** [review]
I did some looking whether there are similar places to look out for, but found nothing.

Also, I would not be opposed to completely remove that trace message: while trace messages may slow down the garbage collection significantly, this message probably does so way too much to be useful for any kind of output; I understand that this message is enabled in development builds only too, but the other collectors do not have similar messages anyway. Additionally, even when debugging it helps much more to have some a bit more targeted log message; this is kind of a last resort to dump *everything*. At that point during debugging adding just another log message is probably not too much of a bother (as you probably have exhausted all other options and littered the code with such messages already. I "know" because I just went through that exercise with another issue in g1, and that single one message wouldn't have helped me in any significant way).

So long story short, if you want to remove that message fine with me, but it might be a good idea to consider other opinions too.

So lgtm.

**Comment by albertnetymk** [comment]
Log tag, `scavenge`, only has `develop_trace` level logs. If they are rarely useful, they should be removed. (I used `scavenge)` to see locate them.)

**Comment by tschatzl** [comment]
Looking at it a bit more, probably all of these `gc, scavenge` log messages should get the same treatment with the local `ResourceMark`. They will otherwise eat tons of memory as every string for every object will be newly allocated in the resource area.

That does not preclude just removing these, but my opinion stands: at the point when you need those, you'll have likely exhausted lots of other avenues already. However maybe this is just my feeling because G1 does not have these, and there you'll need to add messages always. An alternative could be to add similar messages to G1?

It would be good if somebody else could give his opinion.

**Review by lkorinth — CHANGES_REQUESTED** [review]
While the change looks good, I agree with Thomas that it is better to remove the trace altogether. Also, in general it is not always better to move the ResourceMark closer to the allocation. I think you should rename the enhancement to a name that betters describe the new code, and removing the trace (and ResourceMark) is the best way forward.

**Comment by albertnetymk** [comment]
Removed those trace messages as suggested, because the signal-to-noise ratio is too low in them.

**Review by lkorinth — APPROVED** [review]
Looks good to me!

**Review by tschatzl — APPROVED** [review]
Ship it.

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate
